app/services/katello/candlepin/consumer.rb

```diff
@@ -132,11 +132,18 @@ def compliance_reasons
 
       def self.orphaned_consumer_ids
         #returns consumer ids in candlepin with no matching katello entry
-        cp_consumers = ::Katello::Resources::Candlepin::Consumer.get({})
-        cp_consumers.reject! { |consumer| consumer['type']['label'] == 'uebercert' }
-        cp_consumer_ids = cp_consumers.map { |cons| cons["uuid"] }
-        katello_consumer_ids = ::Katello::Host::SubscriptionFacet.pluck(:uuid)
-        cp_consumer_ids - katello_consumer_ids
+        orphaned_ids = []
+        User.as_anonymous_admin do
+          cp_consumers = []
+          Organization.all.each do |org|
+            cp_consumers += ::Katello::Resources::Candlepin::Consumer.get('owner' => org.label)
+          end
+          cp_consumers.reject! { |consumer| consumer['type']['label'] == 'uebercert' }
+          cp_consumer_ids = cp_consumers.map { |cons| cons["uuid"] }
+          katello_consumer_ids = ::Katello::Host::SubscriptionFacet.pluck(:uuid)
+          orphaned_ids = cp_consumer_ids - katello_consumer_ids
+        end
+        orphaned_ids
```

Comment:
yep, in most cases it likely works inside the block, but thats only if you know the code will return the value yielded.  Take for example:

```
def perform_as(user)
  yield
  "#{user} is stupid"
end

def bad_method
  perform_as("Justin") do 
    "Justin is smart!"
  end
end
```

calling 'bad_method' outputs  "Justin is stupid".  While it likely works here, I wouldn't consider the behavior of a method to return the value its yielded block returns as part of a methods 'contract'